x-pack/legacy/plugins/code/server/indexer/lsp_indexer.ts

```diff
@@ -198,14 +211,20 @@ export class LspIndexer extends AbstractIndexer {
   }
 
   protected FILE_OVERSIZE_ERROR_MSG = 'File size exceeds limit. Skip index.';
+  protected BINARY_FILE_ERROR_MSG = 'Binary file detected. Skip index.';
   protected async getFileSource(request: LspIndexRequest): Promise<string> {
-    const { revision, filePath } = request;
-    // Always read file content from the original bare repo
-    const blob = await this.gitOps.fileContent(this.repoUri, filePath, revision);
-    if (blob.rawsize() > TEXT_FILE_LIMIT) {
+    const { filePath } = request;
+    const fullPath = path.join(this.workspaceDir, filePath);
+    const fileStat = await fs.promises.stat(fullPath);
```

Comment:
this `fs.promises` throws some errors though

![image](https://user-images.githubusercontent.com/987855/65548790-0b434580-ded1-11e9-9a81-9709b0c0b269.png)

Reply:
Those are warnings, I don't know if we should ignore this. Normally this warning will be gone if we upgrade node version.